projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/kerberos_plugin.py

```diff
@@ -0,0 +1,121 @@
+                context
+            )
+            try:
+                krb_auth = configuration.get_value(KRB_AUTH)
+                if krb_auth == "minikerberos":
+                    self._authenticator = MinikerberosGSSAPIAuthenticator(
+                        keytab_pathname,
+                        keytab_principal,
+                        keytab_realm,
+                        configuration.get_value(KERBEROS_KDC_HOST),
+                    )
+                elif krb_auth == "kinit":
+                    self._authenticator = KinitGSSAPIAuthenticator(
+                        keytab_pathname,
+                        keytab_principal,
+                    )  # Can kinit the whole process
+                elif krb_auth is None:
+                    self._log.debug("No Kerberos authentication specified")
+                else:
+                    errors.log_and_throw(
+                        to_be_fixed_by=errors.ResolvableBy.CONFIG_ERROR,
+                        log=self._log,
+                        what_happened=f"Provided environment variable {'VDK_KRB_AUTH'} has invalid value.",
+                        why_it_happened=f"VDK was run with environment variable {'VDK_KRB_AUTH'}={krb_auth}, "
+                        f"however '{krb_auth}' is invalid value for this variable.",
+                        consequences=errors.MSG_CONSEQUENCE_DELEGATING_TO_CALLER__LIKELY_EXECUTION_FAILURE,
+                        countermeasures=f"Provide either 'minikerberos' or 'kinit' for environment variable {'VDK_KRB_AUTH'}.",
+                    )
+            except VDKException:
+                self._log.debug(
+                    f"Kerberos authenticator cannot be created. You can provide configuration that specifies "
+                    f"keytab info using the following environment variables: {'VDK_KEYTAB_FOLDER'}, or "
+                    f"{'VDK_KEYTAB_FILENAME'} and {'VDK_KEYTAB_PRINCIPAL'}"
+                )
+                self._authenticator = None
+        return self._authenticator
+
+    @staticmethod
+    def __get_keytab_info(context: JobContext):
+        configuration = context.core_context.configuration
+        job_name = context.name
+        job_directory = str(context.job_directory)
+        keytab_folder = configuration.get_value(KEYTAB_FOLDER)
+        if keytab_folder is None:
+            keytab_folder = job_directory
+        keytab_principal = configuration.get_value(KEYTAB_PRINCIPAL)
+        if keytab_principal is None:
+            keytab_principal = f"pa__view_{job_name}"
+        keytab_filename = configuration.get_value(KEYTAB_FILENAME)
+        if keytab_filename is None:
+            keytab_filename = f"{job_name}.keytab"
+        keytab_pathname = os.path.join(keytab_folder, keytab_filename)
+        keytab_realm = configuration.get_value(KEYTAB_REALM)
+        return keytab_pathname, keytab_principal, keytab_realm
+
+
+@hookimpl
+def vdk_configure(config_builder: ConfigurationBuilder) -> None:
```

Comment:
Let's put teh configuration in a separate file kerberos_config.py and use KerberosConfiguration class which exposes only kerberos configuration. 

See https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_configuration.py